### DISCUSSION TITLE
22 new Nifty Indices added to EOD2

### DISCUSSION THREAD

**Discussion by BennyThadikaran** [BennyThadikaran/eod2#78] [Announcements]
I have added data for 22 new nifty indices from 2015 onwards to current date.

See the [full list of indices synced](https://github.com/BennyThadikaran/eod2_data/blob/main/sector_watchlist.csv) on EOD2. The previous list was too limited and based on my personal choice at the time.

**Let me know, if your prefered Index is missing from this list. I don't mind adding all indexes if many inclusions are being requested.**

```
Nifty Next 50
Nifty 100
Nifty 200
Nifty 500
Nifty Midcap 50
Nifty MNC
Nifty PSE
Nifty Services Sector
Nifty CPSE
India VIX
Nifty Consumer Durables
Nifty India Manufacturing
Nifty Smallcap 250
Nifty Smallcap 50
Nifty MidSmallcap 400
Nifty Midcap 150
Nifty Midcap Select
NIFTY LargeMidcap 250
Nifty India Digital
Nifty Total Market
Nifty Microcap 250
Nifty Healthcare Index
```

**Comment by cotton1234** [comment]
Thanks Benny. Will go through the same as it seems i need to update python on my system to 3.10 and if do upgrade all my automation will break with brokers so creating virtual env to make eod2 module work.

Thanks

**Reply by BennyThadikaran** [reply]
I recently switched to using type annotations in my scripts. Some of them are only supported in Python 3.10. I realised it late, but i can provide backwards compatibility upto 3.8.

**Reply by cotton1234** [reply]
That would be awesome Benny. If its easy for you, pls consider.

**Reply by BennyThadikaran** [reply]
It should not be difficult, just some find and replace (Hopefully!). But i do need to test, if it works correctly on 3.8. I use 3.10. Are you on Windows 10 platform?

**Reply by cotton1234** [reply]
No, i am on Mac Monterey

**Reply by BennyThadikaran** [reply]
I have added compatibility for Python version 3.8 and above. See #79 . I tested it in Docker container with Python 3.8.18 installed.

See [update instructions](https://github.com/BennyThadikaran/eod2/wiki/Installation-for-Linux-and-Mac#updating-the-git-repo) and update `nse` package by running `pip install -U nse`. That should get things working for you.

I would still strongly consider updating to the latest version, since 3.8 will be [end of life](https://devguide.python.org/versions/#supported-versions) by Oct 2024

**Reply by cotton1234** [reply]
I will update by weekend on the same. I am planning to rewrite my automation using the latest python.
Thanks
Saurabh
